lib/validates_overlap/overlap_validator.rb

```diff
@@ -98,11 +102,14 @@ def generate_overlap_sql_conditions(record)
     if record.new_record?
       self.sql_conditions = main_condition
     else
-      self.sql_conditions = "#{main_condition} AND #{record_table_name(record)}.#{primary_key_name} != #{record.send(primary_key_name)}"
+      if primary_key_value(primary_key_name, record).is_a?String
```

Comment:
So right now you're calling `primary_key_value` method 3 times: once on line 105, then on either 106 or 108.  In that case I'd consider creating a local variable to store it, such as:

```
else
  key = primary_key_value(primary_key_name, record)

  if key.is_a?(String)
    # use key here instead of `primary_key_value` method
```